lib/Serialization/DeserializeSIL.cpp

```diff
@@ -598,6 +625,19 @@ SILFunction *SILDeserializer::readSILFunction(DeclID FID,
   if (fn->empty() && errorIfEmptyBody)
     return nullptr;
 
+  // Check that there are no unresolved forward definitions of opened
+  // archetypes.
+  for (auto &KV : OpenedArchetypesTracker.getOpenedArchetypeDefs()) {
+    if (isa<GlobalAddrInst>(KV.getSecond())) {
+      llvm::dbgs() << "The following opened archetype is not resolved: " << KV.getFirst() << "\n";
+      KV.getFirst().dump();
+      fn->verify(true);
+      fn->dump();
+    }
+    assert(KV.getSecond() && !isa<GlobalAddrInst>(KV.getSecond()) &&
```

Comment:
I would suggest using an llvm_unreachable for this error condition. We always want this to fail, even if asserts are disabled.